pkg/kudoctl/cmd/repo_add.go

```diff
@@ -0,0 +1,112 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"strings"
+	"time"
+
+	"github.com/kudobuilder/kudo/pkg/kudoctl/kudohome"
+	"github.com/kudobuilder/kudo/pkg/kudoctl/util/repo"
+
+	"github.com/gofrs/flock"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+)
+
+type repoAddCmd struct {
+	name string
+	url  string
+	home kudohome.Home
+
+	out io.Writer
+	fs  afero.Fs
+}
+
+func (a repoAddCmd) run() error {
+	if err := addRepository(a.fs, a.name, a.url, a.home); err != nil {
+		return err
+	}
+	fmt.Fprintf(a.out, "%q has been added to your repositories\n", a.name)
+	return nil
+
+}
+
+func addRepository(fs afero.Fs, name string, url string, home kudohome.Home) error {
+	f, err := repo.LoadRepositories(fs, home.RepositoryFile())
+	if err != nil {
+		return err
+	}
+	if f.GetRepo(name) != nil {
+		return fmt.Errorf("repository name (%s) already exists, please specify a different name", name)
+	}
+	rc := &repo.RepositoryConfiguration{
+		URL:  url,
+		Name: name,
+	}
+	or, err := repo.NewOperatorRepository(rc)
+	if err != nil {
+		return err
+	}
+	_, err = or.DownloadIndexFile()
+	if err != nil {
+		return fmt.Errorf("looks like %q is not a valid operator repository or cannot be reached: %s", url, err.Error())
+	}
+	// lock file
+	fLock := flock.New(home.RepositoryFile())
+	lockCtx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+	locked, err := fLock.TryLockContext(lockCtx, time.Second)
+	if err == nil && locked {
+		defer fLock.Unlock()
+	}
+	if err != nil {
+		return err
+	}
+	// re-read repositories after lock
+	f, err = repo.LoadRepositories(fs, home.RepositoryFile())
+	if err != nil {
+		return err
+	}
+	f.Add(rc)
```

Comment:
what is f? it's some kind of file? I am really confused about these one char names that just does not correspond to the type or name of the variable